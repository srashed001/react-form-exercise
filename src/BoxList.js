import { useState } from "react";
import {v4 as uuid} from 'uuid'
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";


const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (backgroundColor, width, height) => {
      setBoxes(boxes => [...boxes, {id: uuid(), backgroundColor, width, height}])
  }

  return (
    <>
      <div>
          <NewBoxForm addBox={addBox}/>
      </div>
      <div>
        {boxes.map((box) => (
          <Box key={box.id} color={box.backgroundColor} width={box.width} height={box.height}/>
        ))}
      </div>
    </>
  );
};

export default BoxList;
