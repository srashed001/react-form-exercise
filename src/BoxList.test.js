import { render, fireEvent, getByText } from '@testing-library/react'
import BoxList from './BoxList'

it ("renders without crashing", function(){
    render(<BoxList/>)
})

it("matches snapshot", function(){
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
})

it("renders form in document", function(){
    const {getByTestId} = render(<BoxList />)
    const form = getByTestId('newBoxForm');
    expect(form).toBeInTheDocument();

})

it("can add new box", function(){
    const {queryByTestId, getByLabelText, queryByText} = render(<BoxList />);

    expect(queryByTestId('box')).not.toBeInTheDocument();

    const backgroundColorInput = getByLabelText('Background Color:')
    const widthInput = getByLabelText("Width:")
    const heightInput = getByLabelText("Height:")
    const btn = queryByText('Add Box')

    fireEvent.change(backgroundColorInput, {target: {value: 'red'}})
    fireEvent.change(widthInput, {target: {value: 100}})
    fireEvent.change(heightInput, {target: {value: 100}})
    fireEvent.click(btn)

    expect(queryByTestId('box'))
    .toBeInTheDocument();

})