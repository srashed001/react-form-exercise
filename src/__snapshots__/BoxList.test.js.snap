// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Create a Box
    </h1>
    <form
      data-testid="newBoxForm"
    >
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Background Color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="Width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="Height"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
  <div />
</DocumentFragment>
`;
